src/pypwsqc/flagging.py

```diff
@@ -4,78 +4,95 @@
 
-    Note that this code here is derived from the Python translation,
-    done by Niek van Andel, of the original R code from Lotte de Vos.
-    The Python code stems from here https://github.com/NiekvanAndel/QC_radar.
-    Also note that the correctness of the Python code has not been
-    verified and not all feature of the R implementation might be there.
+    The Python code has been translated from the original R code,
+    to be found here: https://github.com/LottedeVos/PWSQC/tree/master/R.
+
+    The function returns an array with zeros, ones or -1 per time step
+    and station.
+
+    The flag 0 means that no faulty zero has been detected.
+    The flag 1 means that faulty zero has been detected.
+    The flag -1 means that no flagging was done because not enough
+    neighbouring stations are reporting rainfall to make a reliable
+    evaluation.
 
     Parameters
     ----------
-    pws_data
+    pws_data (xr.DataArray)
         The rainfall time series of the PWS that should be flagged
-    reference
+    nbrs_not_nan (xr.DataArray)
+        Number of neighbouring stations reporting rainfall
+    reference (xr.DataArray)
         The rainfall time series of the reference, which can be e.g.
-        the median of neighboring PWS data.
-    nint : optional
+        the median of neighboring stations
+    nint (integer)
         The number of subsequent data points which have to be zero, while
         the reference has values larger than zero, to set the flag for
         this data point to 1.
+    n_stat (integer)
+        Threshold for number of neighbours reporting rainfall
 
     Returns
     -------
-    npt.NDArray
+    xr.DataArray
         time series of flags
     """
-    ref_array = np.zeros(np.shape(pws_data))
-    ref_array[np.where(reference > 0)] = 1
+    # initialize
+    sensor_array = np.empty_like(pws_data)
+    ref_array = np.empty_like(pws_data)
+    fz_array = np.empty_like(pws_data)
 
-    sensor_array = np.zeros(np.shape(pws_data))
+    # Wet timestep at each station
     sensor_array[np.where(pws_data > 0)] = 1
-    sensor_array[np.where(pws_data == 0)] = 0
 
-    fz_array = np.ones(np.shape(pws_data), dtype=np.float_) * -1
+    # Dry timestep at each station
+    sensor_array[np.where(pws_data == 0)] = 0
 
-    for i in np.arange(nint, np.shape(pws_data)[0]):
-        if sensor_array[i] > 0:
-            fz_array[i] = 0
-        elif fz_array[i - 1] == 1:
-            fz_array[i] = 1
-        # TODO: check why `< nint + 1` is used here.
-        #       should `nint`` be scaled with a selectable factor?
-        elif (np.sum(sensor_array[i - nint : i + 1]) > 0) or (
-            np.sum(ref_array[i - nint : i + 1]) < nint + 1
-        ):
-            fz_array[i] = 0
-        else:
-            fz_array[i] = 1
+    # Wet timesteps of the reference
+    ref_array[np.where(reference > 0)] = 1
 
-    # fz_array.data[nbrs_not_nan < nstat] = -1
-    return fz_array
+    for i in np.arange(len(pws_data)):
```

Comment:
I suggest to explicitly use the correct dimension name here since this avoids confusion, i.e. `len(pws_data.id)`.

Anyway, I am not sure if `len(pws_data)` and `len(pws_data[1])` robustly behave in the correct way